corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -14,14 +14,15 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         initialPageData = hqImport("hqwebapp/js/initial_page_data");
 
     // special format handled by CaseSearch API
-    var encodeValue = function (model, value, searchForBlank) {
+    var encodeValue = function (model, searchForBlank) {
+            var value = model.get('value');
             if (value && model.get("input") === "daterange") {
                 value = "__range__" + value.replace(separator, "__");
-            } else if (model.get('input') === 'select') {
+            } else if (value && model.get('input') === 'select') {
                 value = value.join(selectDelimiter);
             }
 
-            var queryProvided = !(value === '' || (_.isArray(value) && _.isEmpty(value)));
+            var queryProvided = _.isObject(value) ? value.length : !!value;
```

Comment:
This logic also needed to be updated now that `value` might be null.

Reply:
Should it be
```suggestion
            var queryProvided = _.isObject(value) ? !!value.length : !!value;
```
I don't quite follow what `value` was before and what it is now - it looks like it might be a string, a list, or an object, depending on the type, is that right?

Reply:
It can be either a string or a list (it'll be a list only for multi-select dropdowns). It won't be an object, so the change from `isArray` to `isObject` isn't necessary, I just liked that the code now _can_ handle an object.